api/envoy/extensions/filters/network/ratelimit/v3/rate_limit.proto

```diff
@@ -32,10 +34,13 @@ message RateLimit {
   // The rate limit domain to use in the rate limit service request.
   string domain = 2 [(validate.rules).string = {min_len: 1}];
 
-  // The rate limit descriptor list to use in the rate limit service request.
+  // The rate limit descriptor list of static values to use in the rate limit service request.
   repeated common.ratelimit.v3.RateLimitDescriptor descriptors = 3
       [(validate.rules).repeated = {min_items: 1}];
 
+  // The rate limit descriptor list of dynamic values to use in the rate limit service request.
+  repeated NetworkRateLimitDescriptor dynamic_descriptors = 8;
```

Comment:
Do we really need a new message here? Can't we just put a substitution format string directly inside the existing descriptor? I admit this could theoretically break someone, but the chance of that in practice is basically zero, and we can runtime feature flag it to be safe?